starport/services/chain/init.go

```diff
@@ -245,18 +233,37 @@ func (c *Chain) Gentx(ctx context.Context, v Validator) (gentxPath string, err e
 	return gentxRe.FindStringSubmatch(gentxPathMessage.String())[1], nil
 }
 
-// AddGenesisAccount add a genesis account in the chain
-func (c *Chain) AddGenesisAccount(ctx context.Context, account spn.GenesisAccount) error {
+// Account represents an account in the chain.
+type Account struct {
+	Name     string
+	Address  string
+	Mnemonic string `json:"mnemonic"`
+	Coins    string
+}
+
+// AddGenesisAccount add a genesis account in the chain.
+func (c *Chain) AddGenesisAccount(ctx context.Context, account Account) error {
+	errb := &bytes.Buffer{}
+
 	return cmdrunner.
 		New(c.cmdOptions()...).
 		Run(ctx, step.New(step.NewOptions().
-			Add(step.Exec(
-				c.app.D(),
-				"add-genesis-account",
-				account.Address.String(),
-				account.Coins.String(),
-			)).
-			Add(c.stdSteps(logAppd)...)...,
+			Add(
+				step.Exec(
+					c.app.D(),
+					"add-genesis-account",
+					account.Address,
+					account.Coins,
+				),
+				step.PostExec(func(exitErr error) error {
+					// ignore if returns with an error related to genesis account being exists.
+					if strings.Contains(errb.String(), "existing") {
```

Comment:
I think we should proceed differently, there could be other error with this keyword and the error message could change in the future

Reply:
Also, I'm not sure if it's a good idea to do this.
Normally the initial genesis and the launch information give us the correct information to generate the genesis. If we have duplicated accounts during genesis generation we should panic because something went wrong with the initial genesis and launch information.